src/common/components/Input/index.tsx

```diff
@@ -138,15 +187,21 @@ export const Input = forwardRef<HTMLInputElement, InputProps>(({ label, error, w
                         </InputLabel>
                     )}
 
-                    <InputElement
-                        id={id}
-                        hasError={indicateError}
-                        aria-invalid={indicateError}
-                        showAlertMessage={showErrorMessage || showWarningMessage}
-                        ref={ref}
-                        {...rest}
-                        onBlur={makeDirty}
-                    />
+                    <Field inputWidth={inputWidth} >
+                        {hide && <IconSpan show={show} onClick={show ? () => setShow(false) : () => setShow(true)}>{show ? viewOn : viewOff}</IconSpan>}
+                        <InputElement
+                            id={id}
+                            hasError={indicateError}
+                            aria-invalid={indicateError}
+                            showAlertMessage={showErrorMessage || showWarningMessage}
+                            ref={ref}
+                            {...rest}
+                            onBlur={makeDirty}
+                            type={show ? null : 'Password'}
```

Comment:
I wonder if there could be a risk with setting the type here as you would override the type that the developers has set when implementing this. Can that happen?